pkg/controller/controller_utils.go

```diff
@@ -758,3 +758,18 @@ func DeleteFinalizer(object metav1.Object, finalizer string) {
 	finalizers.Delete(finalizer)
 	object.SetFinalizers(finalizers.List())
 }
+
+// GetSecretNameFromMachineSetSpec will retrieve the secret name holding AWS credentials
+// from a machcinesetspec if it exists. Otherwise it will return an empty string.
+func GetSecretNameFromMachineSetSpec(msSpec *clusteroperator.MachineSetSpec) (string, error) {
+	secretName := ""
+	if msSpec.ClusterHardware.AWS == nil {
+		return "", fmt.Errorf("no AWS cluster hardware set on machine spec")
+	}
+
+	if msSpec.ClusterHardware.AWS.AccountSecret.Name != "" {
+		secretName = msSpec.ClusterHardware.AWS.AccountSecret.Name
+	}
+
+	return secretName, nil
+}
```

Comment:
Should this return an error if there's no AWS account secret? It looks like that was previous behavior, but I can't think of a reason why it would be anything but an error if that wasn't set, and we went to build a client.

PR looks good to me otherwise.

Reply:
ok. modified to error in the event that there is no secret name provided

Reply:
@dgoodwin so converting to an error has some implications. today the code allows for not defining the AWS account secret name, and it falls back to assuming that there are instance profiles attached to the instance that will allow the AWS API calls. I know we've discussed moving away from instance profiles, but not sure if we're really ready to force creating the AWS creds/secrets. thoughts?

Reply:
Good enough reason to leave it, thanks.